src/Python/qsharp-core/qsharp/tests/test_iqsharp.py

```diff
@@ -82,3 +83,43 @@ def test_multi_compile():
     r = ops[1].simulate()
     assert r == qsharp.Result.One
 
+
+def test_packages():
+    """
+    Verifies default package command
+    """
+    pkg_count = len(qsharp.packages._client.get_packages())
+    assert pkg_count > 0
+
+    with pytest.raises(Exception):
+        qsharp.packages.add('Invalid.Package.!!!!!!')
+    assert pkg_count == len(qsharp.packages._client.get_packages())
+
+    qsharp.packages.add('Microsoft.Extensions.Logging')
+    assert (pkg_count+1) == len(qsharp.packages._client.get_packages())
+
+
+def test_projects():
```

Comment:
In principle, yes... however, deleting the `.csproj` actually leaves the IQ# kernel in a bad state, since it now has a project reference to a non-existent project. Without an API to _remove_ project references, I think it's safer for test purposes to leave the file on disk. I'll change the name of the `.csproj` so that it's more clearly a test artifact.

(It's an interesting question of whether we should expose an API to _remove_ project references, but from a user perspective, this type of thing should be rare and restarting the kernel is pretty easy...)

Reply:
Makes sense. I just always worry a little bit at leaving temp files from a test, but given constraints here I agree that seems like the best solution.

Reply:
I'm changing this to use the `tmp_path` fixture in pytest, which has logic to automatically clean these files up after a number of subsequent runs.